### ISSUE TITLE
[Bug]: Download all files on public share result in a too large archive

### ISSUE THREAD

**Issue by christianlupus** [nextcloud/server#40566]
### ⚠️ This issue respects the following points: ⚠️

- [X] This is a **bug**, not a question or a configuration/webserver/proxy issue.
- [X] This issue is **not** already reported on [Github](https://github.com/nextcloud/server/issues?q=is%3Aopen+is%3Aissue+label%3Abug) OR [Nextcloud Community Forum](https://help.nextcloud.com/) _(I've searched it)_.
- [X] Nextcloud Server **is** up to date. See [Maintenance and Release Schedule](https://github.com/nextcloud/server/wiki/Maintenance-and-Release-Schedule) for supported versions.
- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/).

### Bug description

When using a public share via link of a folder, there is a download option for the complete share. When inside a subfolder of the shared folder, this folder downloads the complete share nevertheless.

### Steps to reproduce

1. Create public share of a folder A which contains files a1, a2, ... and a subfolder B. B contains b1, b2, b3...
2. Navigate in private window to the linked share
3. Navigate to folder B
4. In the top bar, click on the _Download all files_ button
5. Have a look at the downloaded archive

You will see that the files a1,... are there as well as the complete folder B

### Expected behavior

Ony the visible files in the current folder (plus all subfolders of it) is downloaded: b1, ...

### Installation method

None

### Nextcloud Server version

25

### Operating system

None

### PHP engine version

None

### Web server

None

### Database engine version

None

### Is this bug present after an update or on a fresh install?

None

### Are you using the Nextcloud Server Encryption module?

None

### What user-backends are you using?

- [ ] Default user-backend _(database)_
- [ ] LDAP/ Active Directory
- [ ] SSO - SAML
- [ ] Other

### Additional info

Tested on the current instance cloud.nextcloud.com

**Comment by solracsf** [comment]
1. Checkmark the folder
2. Click on Actions
3. Download

?

**Comment by christianlupus** [comment]
Yes, that would work if you knew that the link above isn't doing what is intended. Also this is only possible from the parent folder in a tight way (to keep everything in a folder).
